lib/sarif/gosec_sarif.rb

```diff
@@ -0,0 +1,81 @@
+module Sarif
+  class GosecSarif < BaseSarif
+    GOSEC_URI = 'https://github.com/securego/gosec'.freeze
+
+    def initialize(scan_report)
+      super(scan_report)
+      @uri = GOSEC_URI
+      begin
+        json_obj = JSON.parse(scan_report.log(''))
+        issues = json_obj['Issues']
+        errors = json_obj['Golang errors']
+        parsed_errors = []
+        errors.each do |key|
+          key[1].each do |location|
+            location['uri'] = key[0]
+            parsed_errors << parse_error(location)
+          end
+        end
+        @logs = issues.concat parsed_errors
+      rescue JSON::ParserError
+        @logs = []
+      end
+    end
+
+    def parse_error(error)
+      line = error['line'].to_i
+      column = error['column'].to_i
+      line = 1 if line.zero?
+      column = 1 if column.zero?
```

Comment:
Why set line and column to 1 if they are 0?
I'd keep them 0 to be consistent with the Gosec ouptut.

Reply:
This was because in a file the start position is row 1, column 1. Gosec sometimes returns row 0, column 0 because it couldn't find an exact location for items in "Golang Errors" and this would cause the program to throw an SarifInvalidFormatError. To avoid this i made row 1 column 1, the default position

Reply:
Sounds good1